### PR TITLE
feat: add kubectl auto-completion

### PR DESCRIPTION
**Pull request by AObuchow**
Resolves the last part of https://github.com/eclipse/che/issues/21735 by adding bash autocompletion for kubectl.

To test, use the following image `quay.io/aobuchow/universal-developer-image:kubectl-completion`, i.e.:

```
docker run -ti --rm \
       quay.io/aobuchow/universal-developer-image:kubectl-completion \
       bash
```

Then, with a shell inside the container, type in `kubectl -` and hit TAB twice to see autocomplete results.
The output should resemble the following:

```
projects $ kubectl -
--add-dir-header            --client-key                --log-dir=                  --namespace=                --skip-log-headers          --v=
--alsologtostderr           --client-key=               --log-file                  --password                  --stderrthreshold           --vmodule
--as                        --cluster                   --log-file-max-size         --password=                 --stderrthreshold=          --vmodule=
--as-group                  --cluster=                  --log-file-max-size=        --profile                   --tls-server-name           --warnings-as-errors
--as-group=                 --context                   --log-file=                 --profile-output            --tls-server-name=          -n
--as=                       --context=                  --log-flush-frequency       --profile-output=           --token                     -s
--cache-dir                 --insecure-skip-tls-verify  --log-flush-frequency=      --profile=                  --token=                    -v
--cache-dir=                --kubeconfig                --loglevel                  --request-timeout           --user                      
--certificate-authority     --kubeconfig=               --loglevel=                 --request-timeout=          --user=                     
--certificate-authority=    --log-backtrace-at          --logtostderr               --server                    --username                  
--client-certificate        --log-backtrace-at=         --match-server-version      --server=                   --username=                 
--client-certificate=       --log-dir                   --namespace                 --skip-headers              --v 
```

### CODE REVIEW

File: universal/ubi8/Dockerfile [hunk 1]
```diff
@@ -216,6 +216,10 @@ curl -sSL -o ~/.kubectl_aliases https://raw.githubusercontent.com/ahmetb/kubectl
 echo '[ -f ~/.kubectl_aliases ] && source ~/.kubectl_aliases' >> /home/user/.bashrc
 EOF
 
+# kubectl completion
+RUN kubectl completion bash > /usr/share/bash-completion/completions/kubectl \
+    && echo "source /usr/share/bash-completion/completions/kubectl" >> /home/user/.bashrc
+
```
Comment (l0rd) [inline universal/ubi8/Dockerfile:222]: I would move everything related to kubectl in the dedicated heredoc 

```suggestion
kubectl completion bash > /usr/share/bash-completion/completions/kubectl

echo '[ -f ~/.kubectl_aliases ] && source ~/.kubectl_aliases' >> /home/user/.bashrc
echo "source /usr/share/bash-completion/completions/kubectl" >> /home/user/.bashrc
EOF
```

**Review by l0rd — APPROVED** [review]
A part for a minor suggestion it worked for me

**Review by amisevsk — COMMENTED** [review]
Looking at the .bashrc in the container, I see a warning that sdkman only works if the relevant lines are at the end of .bashrc, but we've injected a bunch of `source` lines after that block -- is the sdkman warning still true? cc: @l0rd 

An easier way to do completions (but one that would require reworking all completions in the dockerfile) would be something like what WTO does:

* In `.bashrc`: `source /etc/profile.d/bash_completion.sh` [[1](https://github.com/redhat-developer/web-terminal-tooling/blob/b0ffeb64ee771c1efd91acd3ff9efd7d27173985/etc/initial_config/.bashrc#L13C1-L13C41)]
* In dockerfile: put completions in `$COMPDIR`: https://github.com/redhat-developer/web-terminal-tooling/blob/b0ffeb64ee771c1efd91acd3ff9efd7d27173985/Dockerfile#L39-L47

**Comment by AObuchow** [comment]
> Looking at the .bashrc in the container, I see a warning that sdkman only works if the relevant lines are at the end of .bashrc, but we've injected a bunch of `source` lines after that block -- is the sdkman warning still true? cc: @l0rd
> 
> An easier way to do completions (but one that would require reworking all completions in the dockerfile) would be something like what WTO does:

Thanks for pointing this out & suggesting an alternative approach @amisevsk. I'm going to take a bit longer to work on this PR to try and see if I can rework all the completions in the dockerfile.

**Comment by l0rd** [comment]
@amisevsk I don't know what's the side effect of the SDKMAN being initialized earlier in the `.bashrc` file, but I agree that we should fix it and your solution is elegant so let's do that.

@AObuchow if you want to do that work in a separate PR is ok too and we can merge this one in the meantime.

**Review by amisevsk — APPROVED** [review]
+1 to merge with @l0rd's suggestion from review, and then optionally rework completions as a separate PR

**Review by amisevsk — APPROVED** [review]

**Comment by AObuchow** [comment]
I think this PR can be closed as it's made redundant by https://github.com/devfile/developer-images/pull/107/files#diff-9f21bef4669d474b6ddef409447be53ae91c94cb50b6df9345149a0e89fb5391R229 @l0rd

**Comment by l0rd** [comment]
> I think this PR can be closed as it's made redundant by [#107 (files)](https://github.com/devfile/developer-images/pull/107/files#diff-9f21bef4669d474b6ddef409447be53ae91c94cb50b6df9345149a0e89fb5391R229) @l0rd

Sorry, that was a mistake. I wanted to test your PR and I rebased mine on top of yours and then I forget about it and pushed your changes too 🤷 . So yes I think we can close this one.

**Comment by AObuchow** [comment]
@l0rd no worries :sunglasses: sounds good, closing.

**Pull request closed without merging**